routes/routes.js

```diff
@@ -0,0 +1,215 @@
+    const product = products.find((product) => product.id === productId);
+    if (!product) return res.status(404).json({ error: 'Product not found' });
+    res.json(product);
+  } catch (error) {
+    res.status(500).json({ error: 'Failed to fetch product' });
+  }
+});
+
+router.put('/cart/:productId', (req, res) => {
+  try {
+    const userId = req.headers['x-user-id'];
+    if (!userId) return res.status(401).json({ error: 'Unauthorized' });
+
+    const productId = parseInt(req.params.productId);
+    const product = products.find((product) => product.id === productId);
+    if (!product) return res.status(404).json({ error: 'Product not found' });
+
+    let cart = carts.find((cart) => cart.userId === userId);
+    if (!cart) {
+      cart = { id: randomUUID(), userId, products: [] };
+      carts.push(cart);
+    }
+    cart.products.push(product);
+    res.status(200).json(cart);
+  } catch (error) {
+    res.status(500).json({ error: 'Failed to update cart' });
+  }
+});
+
+router.delete('/cart/:productId', (req, res) => {
+  try {
+    const userId = req.headers['x-user-id'];
+    if (!userId) return res.status(401).json({ error: 'Unauthorized' });
+
+    const productId = parseInt(req.params.productId);
+    const cart = carts.find((cart) => cart.userId === userId);
+    if (!cart) return res.status(404).json({ error: 'Cart not found' });
+
+    cart.products = cart.products.filter((product) => product.id !== productId);
+    res.status(200).json(cart);
+  } catch (error) {
+    res.status(500).json({ error: 'Failed to remove product from cart' });
+  }
+});
+
+router.post('/cart/checkout', (req, res) => {
+  try {
+    const userId = req.headers['x-user-id'];
+    if (!userId) return res.status(401).json({ error: 'Unauthorized' });
+
+    const cart = carts.find((cart) => cart.userId === userId);
+    if (!cart || cart.products.length === 0)
+      return res.status(400).json({ error: 'Cart is empty' });
+
+    const totalPrice = cart.products.reduce((sum, product) => sum + product.price, 0);
+    const order = { id: randomUUID(), userId, products: cart.products, totalPrice };
+    orders.push(order);
+
+    cart.products = [];
+    res.status(201).json(order);
+  } catch (error) {
+    res.status(500).json({ error: 'Failed to create order' });
+  }
+});
+
+router.post('/product', (req, res) => {
+  const { error } = productSchema.validate(req.body);
+  if (error) return res.status(400).json({ error: error.details[0].message });
+
+  const newProduct = {
+    id: randomUUID(),
+    ...req.body,
+  };
+
+  products.push(newProduct);
+  fs.writeFileSync('storage/products.store.json', JSON.stringify(products, null, 2));
+  res.status(201).json({ message: 'The product has been created successfully' });
+});
+
+router.post('/products/import', upload.single('file'), (req, res) => {
+  try {
+    const results = [];
+    eventEmitter.emit('fileUploadStart');
+
+    fs.createReadStream(req.file.path)
```

Comment:
check the example of using Multer with stream in this article 
[https://devsarticles.com/multer-file-upload-nodejs-complete-guide](https://devsarticles.com/multer-file-upload-nodejs-complete-guide)